app/sidekiq/generate_sales_report_job.rb

```diff
@@ -4,27 +4,35 @@ class GenerateSalesReportJob
   include Sidekiq::Job
   sidekiq_options retry: 1, queue: :default, lock: :until_executed, on_conflict: :replace
 
-  def perform(country_code, start_date, end_date, send_notification = true, s3_prefix = nil)
+  ALL_SALES = "all_sales"
+  DISCOVER_SALES = "discover_sales"
+  SALES_TYPES = [ALL_SALES, DISCOVER_SALES]
+
+  def perform(country_code, start_date, end_date, sales_type, send_slack_notification = true, s3_prefix = nil)
     country = ISO3166::Country[country_code].tap { |value| raise ArgumentError, "Invalid country code" unless value }
+    raise ArgumentError, "Invalid sales type" unless SALES_TYPES.include?(sales_type)
 
-    start_time_of_quarter = Date.parse(start_date.to_s).beginning_of_day
-    end_time_of_quarter = Date.parse(end_date.to_s).end_of_day
+    start_time = Date.parse(start_date.to_s).beginning_of_day
```

Comment:
Renamed these because they are not specific to `quarter`s I think and can be a generic time range.